src/svc_rqst.c

```diff
@@ -154,12 +154,16 @@ svc_rqst_lookup_chan(uint32_t chan_id)
 	if (chan_id >= svc_rqst_set.max_id)
 		return (NULL);
 
+	mutex_lock(&svc_rqst_set.mtx);
```

Comment:
So, this causes a problem, and so far I haven't been able to find a way around it.  svc_rqst_set.mtx needs to be taken in svc_rqst_release(), as well, but that will cause a lock inversion with this because it already has sr_rec->mtx taken when it's called.  See the comment there, as well.

Reply:
I'm not sure this is even useful. Did it stop the crash?

In my original thought, only need to lock to set, the get is atomic.

But once somebody "owns" the sr_rec itself, nobody else can clear it. The outer lock and refcount prevents it.